### PR TITLE
Logger & better errors handling

### PR DESCRIPTION
**Pull request by FrancoisPog**
## The `Logger`
To centralize message logging, the Logger provides 3 methods: `info`, ` debug` and `error`. When `NODE_ENV` is set to `production`, messages are printed in the `out.txt` and `error.txt` files, in the `logs` directory. Otherwise, they are printed on the console

## Errors handling
In asynchronous chain promises, detected errors should be rejected and handled (and logged) by the top function.
To keep the context of the error being triggered, the function should reject the error after adding a title with the `addTitleError` function (like `perror`).

### In controllers
In the controllers, we are faced with two types of errors, usage error or server error. 
The first is caused by improper use from the user, in this case we have to send the error to the user. 
The second type is caused by unexpected failures (db query, etc), in this case we need to send an error back to the user but hide the error and log it to the server.

To easily handle this, the `HttpResponseWrapper` provides 3 methods, `sendResponse`, `sendUsageError` and `sendServerError`.

#### Controllers conventions 
The `httpResponseWrapper` should only be managed by the controller itself and not by subfunctions, as it is not their role to deal with http responses.
About this, we should follow these rules, in functions returning promises, we need to reserve rejection if we can't do processing, for example if we try to retrieve information about non-existent user data, we need to resolve to null, but not reject.
Conversely, if a database query fails, we should reject the promise.

### CODE REVIEW

**Review by nathanaelhoun — COMMENTED** [review]
Great work, the code is really better like this!
Can the `HttpResponseWrapper` be used as a middleware to DRY the code?

**Comment by nathanaelhoun** [comment]
Heads up that @FrancoisPog you have some place to update from #144 :smile:

File: server/global/HttpResponseWrapper.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    this.sendUsageError = this.sendUsageError.bind(this);
+    this.sendResponse = this.sendResponse.bind(this);
+  }
+
+  /**
+   * Send the error
+   * @param {Error|UsageError} error
+   */
+  sendServerError() {
+    return this.sendResponse(500, { message: "Server side error" });
```
[the remark this answers is not in the document]
Comment (FrancoisPog) [reply server/global/HttpResponseWrapper.js:20]: of course yes 😄

File: server/global/ErrorLogger.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+ * Log an error
+ * @param {Error} error The error object
+ * @param {string} message Message explaining the reason for the error
+ */
+export function logError(error, message = "") {
+  const line = Array(100).fill("-").join("");
+  const txt = `\n${line}\n${new Date()} - ${message || "No message provided"}\n${line}\n> ${
+    error.stack
+  }\n${line}`;
+  console.error(txt);
```
[the remark this answers is not in the document]
Comment (FrancoisPog) [reply server/global/ErrorLogger.js]: yep it's done !

File: server/global/ErrorLogger.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Log an error
+ * @param {Error} error The error object
+ * @param {string} message Message explaining the reason for the error
+ */
+export function logError(error, message = "") {
+  const line = Array(100).fill("-").join("");
```
[the remark this answers is not in the document]
Comment (FrancoisPog) [reply server/global/ErrorLogger.js]: Not necessary, but it formats the error a bit

File: server/db/database.js [hunk 4]
```diff
@@ -19,120 +18,155 @@ const versions = [
...
+ * Get the current API version
+ * @returns {string} The version "YYYY-MM-DD"
+ */
+export const currentAPIVersion = () => versions[versions.length - 1];
 
-Database.connection = mysql.createConnection({
+export const connection = mysql.createConnection({
   host: process.env.DB_HOST || "localhost",
-  user: process.env.DB_USER || "glowing-octo-guacamole",
+  user: process.env.DB_UdER || "glowing-octo-guacamole",
```
Comment (valentinperignon) [inline server/db/database.js:29]: Is it a typo?

**Review by valentinperignon — APPROVED** [review]
LGTM!!

File: server/db/database.js (hunk 4 above)
Comment (FrancoisPog) [reply server/db/database.js:29]: yes

**Review by nathanaelhoun — APPROVED** [review]
Awesome :rocket:

**Pull request merged**